### PR TITLE
Add builtin function ord

### PR DESCRIPTION
**Pull request by ts25504**
Add built-in function 'ord' in builtin_string.go

### CODE REVIEW

File: expression/builtin_string.go [hunk 1]
```diff
@@ -1506,7 +1506,37 @@ type builtinOrdSig struct {
...
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	if len(str) == 0 {
+		d.SetInt64(0)
+		return d, nil
+	}
+
+	leftMost := string([]rune(str)[0])
```
Comment (coocood) [inline expression/builtin_string.go]: Use `utf8.DecodeRuneInString` can avoid memory allocation.

**Comment by ts25504** [comment]
@coocood I just revised my implement using `utf8.DecodeRuneInString` instead, please check it.

File: expression/builtin_string.go [hunk 2]
```diff
@@ -1506,7 +1507,38 @@ type builtinOrdSig struct {
...
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	if len(str) == 0 {
+		d.SetInt64(0)
+		return d, nil
+	}
+
+	leftMost, _ := utf8.DecodeRuneInString(str)
```
Comment (coocood) [inline expression/builtin_string.go]: use the second return value to slice the string, then we don't need to convert it to bytes.

**Comment by ts25504** [comment]
@coocood Hello! I followed your advice and used the second return value to slice the string. Is that correct?

**Comment by coocood** [comment]
LGTM
Thank you!

File: plan/typeinferer.go [hunk 3]
```diff
@@ -412,6 +412,8 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
 		tp.Flen = 40
 	case ast.Coalesce:
 		tp = aggArgsType(x.Args)
+	case ast.Ord:
```
Comment (zimulala) [inline plan/typeinferer.go]: You can put this case to line 383.

Comment (ts25504) [inline plan/typeinferer.go]: @zimulala Hi! I just move this case statement to another place, please review it.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**